src/destinations/MySqlInsert.ts

```diff
@@ -0,0 +1,263 @@
+    query: string,
+    queryValues: Array<string | number>,
+}
+
+export interface MySqlInsertResult {
+    fieldCount: number,
+    affectedRows: number,
+    insertId: number,
+    serverStatus: number,
+    warningCount: number,
+    message: string,
+    protocol41: boolean,
+    changedRows: number,
+}
+
+export interface MySqlUpdateResult {
+    fieldCount: number,
+    affectedRows: number,
+    insertId: number,
+    serverStatus: number,
+    warningCount: number,
+    message: string,
+    protocol41: boolean,
+    changedRows: number,
+}
+
+export interface MySqlDeleteResult {
+    fieldCount: number,
+    affectedRows: number,
+    insertId: number,
+    serverStatus: number,
+    warningCount: number,
+    message: string,
+    protocol41: boolean,
+    changedRows: number,
+}
+
+export interface MySQLConfig {
+    tableName: string,
+    bulkDeleteMatchFields: Array<string>,
+    deleteAllRecordsBeforeLoad: boolean,
+}
+
+export class MySqlInsert extends EtlDestination {
+    protected mySqlClient: DBClient;
+    protected tableName: string;
+    protected bulkDeleteMatchFields: Array<string>;
+    protected deleteAllRecordsBeforeLoad: boolean;
+
+    /**
+     * Upload data to MySQL database table
+     */
+    constructor(mySqlClient: DBClient, tableDetails: MySQLConfig) {
+        super();
+        this.mySqlClient = mySqlClient;
+        this.tableName = tableDetails.tableName;
+        this.bulkDeleteMatchFields = tableDetails.bulkDeleteMatchFields;
+        this.deleteAllRecordsBeforeLoad = !tableDetails.bulkDeleteMatchFields && tableDetails.deleteAllRecordsBeforeLoad;
+    }
+
+    public getMySqlClient(): DBClient {
+        return this.mySqlClient;
+    }
+
+    public getTableName(): string {
+        return this.tableName;
+    }
+
+    public getBulkDeleteMatchFields(): Array<string> {
+        return this.bulkDeleteMatchFields;
+    }
+
+    public getDeleteAllRecordsBeforeLoad(): boolean {
+        return this.deleteAllRecordsBeforeLoad;
+    }
+
+    /**
+     * Stores the batch records in a file
+     * @param batch
+     */
+    public async store(batch: EtlBatch): Promise<void> {
+        try {
+            await this.processRecords(batch);
+            log.debug(`finish uploading: ${batch.getBatchNumber()}`);
+        } catch (error) {
+            await batch.setState(EtlState.ERROR);
+            throw(error);
+        }
+    }
+
+    public async processRecords(batch: EtlBatch): Promise<boolean> {
+        const fieldList = batch.getTransformedRecords().map( (record) => record.getTransformedData() );
+        const deleteAllRecords = this.deleteAllRecordsBeforeLoad && (batch.getBatchNumber() === 1);
+        return await this.mySqlClient.runInTransaction(false, async (transaction: DBTransaction) => {
+            if (deleteAllRecords) {
+                log.debug(`Delete all records from table : ${this.tableName}`);
+                await transaction.query(`delete from ${this.tableName}`);
```

Comment:
whilst this works, truncating the table is much faster.
so depending on how many records you are deleting, may be worth doing that instead

Reply:
True, but we can't truncate in most of the databases because only the root user has permission to truncate tables